crates/matrix-sdk-ui/src/timeline/event_handler.rs

```diff
@@ -302,9 +304,20 @@ impl<'a> TimelineEventHandler<'a> {
                 }
             },
 
-            TimelineEventKind::RedactedMessage => {
-                self.add(NewEventTimelineItem::redacted_message());
-            }
+            TimelineEventKind::RedactedMessage { event_type } => match event_type {
+                MessageLikeEventType::Reaction => {
+                    if let Flow::Remote { position: TimelineItemPosition::Update(idx), .. } =
+                        self.flow
+                    {
+                        // Filter out redacted reactions from the timeline.
+                        self.items.remove(idx);
+                        self.result.item_removed = true;
+                    }
+                }
+                _ => {
+                    self.add(NewEventTimelineItem::redacted_message());
+                }
+            },
```

Comment:
We shouldn't have a removal branch. Simply not adding an item when `Flow::Remote { position: TimelineItemPosition::Update(_), .. }` is handled will already result in the existing UTD removal branch being taken.

```suggestion
            TimelineEventKind::RedactedMessage { event_type } => {
                if event_type != MessageLikeEventType::Reaction {
                    self.add(NewEventTimelineItem::redacted_message());
                }
            },
```